test/agent/cmd/aws-node-config/main.go

```diff
@@ -0,0 +1,156 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package main
+
+import (
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/agent/pkg/input"
+)
+
+const (
+	MTU_CHECK         = "mtu"
+	VETH_PREFIX_CHECK = "veth"
+)
+
+type MTU_CONFIG struct {
+	ExpectedVal int
+}
+
+func main() {
+	validate := ""
+	config := ""
+
+	flag.StringVar(&validate, "validate", "", "name of the test, valid values: mtu, veth")
+	flag.StringVar(&config, "config", "", "Base64 encoded json object for the given test")
+	flag.Parse()
+
+	decoded, err := base64.StdEncoding.DecodeString(config)
+	if err != nil {
+		log.Fatalf("failed to descode base64 string %s: %v", config, err)
+	}
+
+	switch validate {
+	case MTU_CHECK:
+		validateMTUValue(decoded)
+	case VETH_PREFIX_CHECK:
+		validateVethPairCount(decoded)
```

Comment:
I think this can already be done using host networking binary. Did you get a chance to see if that can be used?

Reply:
Yes, I did, I thought we wanted to keep this binary separate. Do you want me to move this logic to host networking binary instead ?